Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Receivers/InteractionReceiver.cs

```diff
@@ -0,0 +1,410 @@
+using Microsoft.MixedReality.Toolkit.Core.Interfaces.InputSystem;
+using Microsoft.MixedReality.Toolkit.Core.Managers;
+using Microsoft.MixedReality.Toolkit.SDK.Input;
+using Microsoft.MixedReality.Toolkit.SDK.Input.Handlers;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX.Receivers
+{
+    /// <summary>
+    /// An interaction receiver is simply a component that attached to a list of interactable objects and does something
+    /// based on events from those interactable objects.  This is the base abstract class to extend from.
+    /// </summary>
+    public abstract class InteractionReceiver : BaseInputHandler,
+        IMixedRealityFocusChangedHandler,
+        IMixedRealityGestureHandler,
+        IMixedRealityInputHandler,
+        IMixedRealityGestureHandler<Vector2>,
+        IMixedRealityGestureHandler<Vector3>,
+        IMixedRealityGestureHandler<Quaternion>
+
+    {
+        #region Public Members
+        /// <summary>
+        /// List of linked interactable objects to receive events for
+        /// </summary>
+        [SerializeField]
+        [ Tooltip("Target interactable Object to receive events for")]
+        private List<GameObject> interactables = new List<GameObject>();
+
+        public List<GameObject> Interactables
+        {
+            get { return interactables; }
+            set { value = interactables; }
+        }
+
+        /// <summary>
+        /// List of linked targets that the receiver affects
+        /// </summary>
+        [Tooltip("Targets for the receiver to ")]
+        private List<GameObject> targets = new List<GameObject>();
+
+        public List<GameObject> Targets
+        {
+            get { return targets; }
+            set { value = targets; }
+        }
+
+        [Tooltip( "If true, this object will remain the prime focus while select is held" )]
+        [SerializeField]
+        private bool lockFocus = false;
+
+        /// <summary>
+        /// Flag for locking focus while selected
+        /// </summary>
+        public bool LockFocus
+        {
+            get
+            {
+                return lockFocus;
+            }
+            set
+            {
+                lockFocus = value;
+                CheckLockFocus(_selectingFocuser);
+            }
+        }
+        
+        #endregion
+
+        #region Private and Protected Members
+
+        /// <summary>
+        /// Protected focuser for the current selecting focuser
+        /// </summary>
+        protected IMixedRealityPointer _selectingFocuser;
+        
+        #endregion
+
+        /// <summary>
+        /// On start subscribe to all interaction events on elements in the interactables list.
+        /// </summary>
+        protected override void OnEnable()
+        {
+            base.OnEnable();
+            //Still need to register since focus isn't required
+            if(InputSystem != null && InputSystem.IsInputEnabled)
+            {
+                InputSystem.Register( gameObject );
```

Comment:
The base input handler already does this kind of stuff. We could make `IsFocusRequired` virtual in the base input handler and override it in this class, then just call the base.

Reply:
This is what you mentioned yesterday. That sounds much more elegant. I can add it to this PR.

Reply:
Since `IsFocusedRequired` is a public accessor, can't we just set it to false before we call `base.onEnable()`? Do we need to override the accessor here?

Reply:
We could make the `IsFocusedRequired`'s set accessor protected